axelor-sale/src/main/java/com/axelor/apps/sale/web/SaleOrderController.java

```diff
@@ -737,4 +738,14 @@ public void recoverVersion(ActionRequest request, ActionResponse response)
     }
     response.setReload(true);
   }
+
+  public void updateRelatedOrderLines(ActionRequest request, ActionResponse response)
+      throws AxelorException {
+    SaleOrder saleOrder = request.getContext().asType(SaleOrder.class);
+    if (Beans.get(AppSaleService.class).getAppSale().getIsSubLinesEnabled()) {
```

Comment:
The condition should be done inside the method of the service, not in the controller.